File: MensLacrosseStats/View Controllers/CollectionViewCells/INSOPlayerCollectionViewCell.m
//
//  INSOPlayerCollectionViewCell.m
//  MensStatsTracker
//
//  Created by James Dabrowski on 9/28/15.
//  Copyright © 2015 Intangible Software. All rights reserved.
//

#import "UIColor+INSOScorebookColor.h"

#import "INSOPlayerCollectionViewCell.h"

@implementation INSOPlayerCollectionViewCell

- (void)awakeFromNib
{
    // Taken from: http://pinkstone.co.uk/how-to-build-a-uicollectionview-in-ios-8/
    // standard background (deselected)
    UIView *bgView = [[UIView alloc]initWithFrame:self.bounds];
    self.backgroundView = bgView;
    self.backgroundView.backgroundColor = [UIColor whiteColor];
    self.backgroundView.layer.borderColor = [[UIColor scorebookBlue] CGColor];
    self.backgroundView.layer.borderWidth = 1.0;
    
    // selected background
    UIView *selectedView = [[UIView alloc]initWithFrame:self.bounds];
    self.selectedBackgroundView = selectedView;
    self.selectedBackgroundView.backgroundColor = [UIColor scorebookBlue];
    self.selectedBackgroundView.layer.borderColor = [[UIColor whiteColor] CGColor];
    self.selectedBackgroundView.layer.borderWidth = 1.0;
}

@end

File: MensLacrosseStats/View Controllers/CollectionViewCells/INSOPlayerCollectionViewCell.swift
//  Converted to Swift 5.5 by Swiftify v5.5.27463 - https://swiftify.com/
//
//  INSOPlayerCollectionViewCell.swift
//  MensStatsTracker
//
//  Created by James Dabrowski on 9/28/15.
//  Copyright © 2015 Intangible Software. All rights reserved.
//

import UIKit

class INSOPlayerCollectionViewCell: UICollectionViewCell {
    @IBOutlet weak var playerNumberLabel: UILabel!
    @IBInspectable var highlightColor: UIColor?

    override func awakeFromNib() {
        super.awakeFromNib()

        backgroundColor = UIColor.white
        layer.borderColor = highlightColor?.cgColor
        layer.borderWidth = 1.0
    }
    
    override var isSelected: Bool {
        didSet {
            if self.isSelected {
                UIView.animate(withDuration: 0.2) {
                    self.backgroundColor = self.highlightColor
                }
            } else {
                UIView.animate(withDuration: 0.2) {
                    self.backgroundColor = UIColor.white
                }
            }
        }
    }
}
